docker/maistra-builder_2.5.Dockerfile

```diff
@@ -1,229 +1,446 @@
-FROM quay.io/centos/centos:stream8
+# Copyright Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
-# Versions
+# hadolint ignore=DL3006
+FROM registry.access.redhat.com/ubi9/ubi:9.2-696
+
+ARG TARGETARCH
+
+RUN set -eux; \
```

Comment:
Setting `TARGETARCH` in this `RUN` statement has no effect beyond this `RUN` statement.
For the value to be accessible throughout the Dockerfile, it needs to be set by passing a `--build-arg TARGETARCH=<amd64|arm64>` on the `docker build` command line.

Reply:
This approach was from upstream istio/tools
ref: https://github.com/istio/tools/blob/master/docker/build-tools/Dockerfile

The TARGETARCH is a predefined build ARG. That is supplied with automatically injected values. So we don't need to specifically pass  --build-arg argument in a docker build command line.

> Docker supports some build arguments by default, even if you don't include their ARG instructions in your Dockerfile. They [relate to proxy settings](https://docs.docker.com/engine/reference/builder/#predefined-args) and work whenever their corresponding --build-arg flag is used. The variables are also excluded from docker history output to avoid disclosing the potentially sensitive details they're intended for - more on this command and its implications below.

 > Builds handled by the [BuildKit backend](https://www.howtogeek.com/devops/what-is-dockers-buildkit-and-why-does-it-matter/) can access several other predefined build args too. These are supplied with automatically injected values. [The list](https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope) includes TARGETOS, TARGETARCH, TARGETPLATFORM, and BUILDOS, BUILDARCH, and BUILDPLATFORM, among several others. The variables describe characteristics of the build environment and the platform the new image is targeting.

Reply:
sorry, I misunderstood our CI make target. I see now. Our CI node is x86 only so there is no ARM builder image being built. I will add a new make target in the makefile.